src/main/java/com/hierynomus/mssmb2/SMB2OplockBreakLevel.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C)2016 - SMBJ Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hierynomus.mssmb2;
+
+import com.hierynomus.protocol.commons.EnumWithValue;
+
+/**
+ * [MS-SMB2].pdf 2.2.23 SMB2 OPLOCK_BREAK Notification - OplockLevel
+ */
+public enum SMB2OplockBreakLevel implements EnumWithValue<SMB2OplockBreakLevel> {
```

Comment:
This is because the valid set is different between requesting oplock and breaking oplock. Only `LEVEL_NONE` and `LEVEL_II` are **valid for breaking oplock**. Base on my understanding, each enum class should only contain the valid values (and also null in Java). Adding `valid check` functions maybe is an alternative but `valid check` function check on **run time**. Separating this enum will force the checking on **compile time**. In my opinion, error checking on compile time is better than on run time. For me, I think simply separate this enum should make the code looks simpler than calling `valid check` function.

Reply:
Please model it according to spec, there is no `SMB2_OPLOCK_BREAK_LEVEL`, only `SMB2_OPLOCK_LEVEL`. With different enums we need to have translation code in place to transform one into the other. Please merge them.